File: src/spirite/base/pen/behaviors/PenBehavior.java
package spirite.base.pen.behaviors;

import spirite.base.brains.ToolsetManager.PenDrawMode;
import spirite.base.brains.ToolsetManager.Tool;
import spirite.base.brains.ToolsetManager.ToolSettings;
import spirite.base.pen.Penner;
import spirite.base.pen.StrokeEngine;
import spirite.base.pen.StrokeEngine.StrokeParams;

public class PenBehavior extends StrokeBehavior {
	final int color;
	public PenBehavior( Penner penner, int i) {
		super(penner);
		this.color = i;
	}
	@Override
	public void start() {
		ToolSettings settings = this.penner.toolsetManager.getToolSettings(Tool.PEN);
		StrokeEngine.StrokeParams stroke = new StrokeEngine.StrokeParams();
		stroke.setColor( color);
		stroke.setMode((PenDrawMode)settings.getValue("mode"));
		stroke.setWidth((float)settings.getValue("width"));
		stroke.setAlpha((float)settings.getValue("alpha"));
		stroke.setHard((Boolean)settings.getValue("hard"));
		
		// Start the Stroke
		startStroke( stroke);
	}
}

File: src/main/kotlin/spirite/base/pen/behaviors/PenBehavior.kt
package spirite.base.pen.behaviors

import rb.glow.Color
import spirite.base.brains.toolset.IToolsetManager
import spirite.base.imageData.drawer.IImageDrawer.IStrokeModule
import spirite.base.pen.Penner
import spirite.base.pen.stroke.StrokeParams
import spirite.base.pen.stroke.StrokeParams.Method


object PenBehavior {
    class Stroke(penner: Penner, drawer: IStrokeModule, val color: Color) : StrokeBehavior(penner,drawer) {
        override fun makeStroke() = makeStroke(penner.toolsetManager, color)
    }

    private fun makeStroke(toolsetManager: IToolsetManager, color: Color) : StrokeParams{
        val settings = toolsetManager.toolset.Pen

        return StrokeParams(
                color,
                Method.BASIC,
                settings.mode,
                settings.width,
                settings.alpha,
                hard = settings.hard)
    }
}